product_multi_company/tests/test_product_multi_company.py

```diff
@@ -124,15 +124,19 @@ def test_search_product(self):
         )
         searched_templates = self.env["product.template"].search(
             [
-                ("company_id", "in", [self.company_1.id, False]),
                 ("id", "in", expected_products.product_tmpl_id.ids),
+                "|",
+                ("company_id", "=", self.company_1.id),
+                ("company_id", "=", False),
```

Comment:
You can't change the test condition to fit your needs. Take a look on why this domain shape needs to be tested: [b240644](https://github.com/OCA/multi-company/pull/726/commits/b240644610a9be35db296ff0b3742902c1149abf)